src/sk_transformers/generic_transformer.py

```diff
@@ -192,16 +191,73 @@ class AggregateTransformer(BaseTransformer):
     ```
 
     Args:
-        features (List[Tuple[str, str, List[str]]]): List of tuples containing the column identifiers and the aggregation function(s).
-                The first column identifier (features[0]) is the column that will be used to group the data.
-                It can be either numerical or categorical. The second column identifier (features[1]) is the column that will be used
-                for aggregations. This column must be numerical.
+        features (List[Tuple[str, Tuple[str, Union[str, Callable], str]]]): List of tuples where
+            the first element is the name or the list of names of columns to be grouped-by,
+            and the second element is a tuple or list of tuples containing the aggregation information.
+            In this tuple, the first element is the name of the column to be aggregated,
+            the second element is the aggregation function as a string or function object,
+            and the third element is the name of the new aggregated column.
     """
 
-    def __init__(self, features: List[Tuple[str, str, List[str]]]) -> None:
+    def __init__(
+        self,
+        features: List[
+            Tuple[
+                Union[str, List[str]],
+                Union[
+                    Tuple[str, Union[str, Callable], str],
+                    List[Tuple[str, Union[str, Callable], str]],
+                ],
+            ]
+        ],
+    ) -> None:
         super().__init__()
         self.features = features
 
+    def __get_list_of_features(self) -> List[str]:
+        feature_list = []
+
+        for groupby_columns, agg_features in self.features:
+            if isinstance(groupby_columns, str):
+                groupby_columns = [groupby_columns]
+            for groupby_column in groupby_columns:
+                feature_list.append(groupby_column)
+
+            if isinstance(agg_features, tuple):
+                agg_features = [agg_features]
+            for agg_column, _, _ in agg_features:
+                feature_list.append(agg_column)
+
+        return list(set(feature_list))
+
+    def __check_input_type(self) -> None:
```

Comment:
Sorry for one more comment. But this function name is not correct, right? The function is not really checking the input types but checking the length of the feature. 

Not sure if this refers to: https://github.com/chrislemke/sk-transformers/pull/79#discussion_r1098641879

Reply:
Yes, it refers to [#79 (comment)](https://github.com/chrislemke/sk-transformers/pull/79#discussion_r1098641879).

The function partly checks for the type, but also does other checks. How about the name `__check_input()`?

Reply:
I like it!